src/key_dispatch.rs

```diff
@@ -0,0 +1,440 @@
+use crate::cursor::CursorMove;
+use crate::input::{Input, Key};
+use crate::scroll::Scrolling;
+use crate::TextArea;
+pub(crate) fn input(ta: &mut TextArea<'_>, input: Input) -> bool {
+    // Should we start selecting text, stop the current selection, or do nothing?
+    // the end is handled after the ending keystroke
+    let end_select = match (ta.select_start.is_some(), input.shift) {
```

Comment:
good catch

Reply:
actually not so, the character input closes the selection

```
    pub fn insert_char(&mut self, c: char) {
        self.delete_selection();
```

so that 'A' replaces whatever was selected, which is the desired result